### DISCUSSION TITLE
DAW Recording light

### DISCUSSION THREAD

**Discussion by dinoslavr** [tttapa/Control-Surface#451] [Q&A]
Hi Pieter, thank you for this great library!

I have used it to build a transport control with an Arduino Mega compatible board, to which I have uploaded the mocoLufa software and it works great! 
What I haven't been able to achieve yet is add to it the lines of code that will light up an LED and change the state of a relay, that will light a bulb when recording starts on Logic Pro x. I know what MIDI messages Logic sends when recording is enabled, but since I'm new to programming, I haven't been able to read them using your library and change the state of pins on my board.
Maybe this interests other users as well.
Your advice on this would be much appreciated! 

Regards
Dinos

**Comment by tttapa** [comment]
> I know what MIDI messages Logic sends when recording is enabled

If you share that information here, I might be able to tell you how to receive it with Control Surface. You can probably use something like the [`NoteLED`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d8/d92/1_8Note-LED_8ino-example.html) example, but I'm not familiar with Logic's MIDI implementation, so I can't answer that definitively without knowing the actual MIDI messages.

**Reply by dinoslavr** [reply]
Thank you for your reply! I have tried to use the NoteLED example, but when I replace the note letter (C)  with a number (25), I get an error in compilation. Logic sends a note on message on channel 2 when recording starts (note 25, velocity 127) and another one when recording stops (note 25, velocity 0). When a track is armed for recording, Logic sends note on with note number 24, velocity 127 and when no track is armed the velocity of note 24 changes to 0. There is also a built in control surface in Logic for this exact purpose. The only parameters that can be changed are the midi in and out ports, the midi status and the colour. All other parameters are not user defined. I'm attaching screenshots of the settings window for the recording light control surface. 

![Screen Shot 2021-05-14 at 11 41 35](https://user-images.githubusercontent.com/82338269/118245552-a4a9db80-b4a9-11eb-880c-9165def6a4c5.png)

![Logic](https://user-images.githubusercontent.com/82338269/118245096-04ec4d80-b4a9-11eb-8bee-103ebdd8de5f.png)

**Reply by tttapa** [reply]
> when I replace the note letter (C) with a number (25), I get an error in compilation

Hard to tell without seeing your code, but if you replaced `note(C, 4)` with `note(25)`, that won't work. You have to use:
```cpp
NoteValueLED led = {
  LED_BUILTIN,     // Pin of built-in LED
  {25, CHANNEL_2}, // Note 25 on MIDI channel 2
};
```
This page might be useful: [MIDI Addresses](https://tttapa.github.io/Control-Surface-doc/Doxygen/d6/dc6/md_pages_Basics.html)

**Reply by dinoslavr** [reply]
Thank you! It works!! 
Here is the code I used, in case anyone else finds it useful!
(This in only for Logic Pro!!!)
Thank you again for your time and this great library!!


#include <Control_Surface.h> // Include the Control Surface library

// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;

using namespace MIDI_Notes;

/* Instantiate the relay that will "blink" when one or more tracks are armed for recording and remain closed while recording is enabled */

NoteValueLED relay = {
  46,                      // Pin where relay is connected
  {24, CHANNEL_2},         // Note 24 on MIDI channel 2
};

void setup() {
  Control_Surface.begin(); // Initialize Control Surface
}

void loop() {
  Control_Surface.loop(); // Update the Control Surface
}

**Comment by benwadub** [comment]
hi @dinoslavr where did you find how to send midi from logic to controller please?

**Reply by dinoslavr** [reply]
HI, I'm not sure I understand what you're asking. Logic sends MIDI signals by default, all you have to do is read them using the code, is that your question or have I misunderstood?

**Reply by Fede-VR73** [reply]
@benwadub  : 
Midi Environment is where  in logic u can manage midi in a deep way,  have a look there : 
https://support.apple.com/guide/logicpro/environment-overview-lgcpf755dc51/mac
;-)

**Reply by benwadub** [reply]
@dinoslavr my question is to know what midi information send by exemple when track 1 is muted I want to know what midi cc is sent or even better set the midi information I need to light my led.
@Fede-VR73 i ll try to look at it but I never used it! hope I ll understand it!

**Comment by dinoslavr** [comment]
Maybe you could find useful information about that in Controller Assignments window, under Control Surfaces menu in Logic Pro

**Reply by benwadub** [reply]
I looked at it but seems to be only input Information, I can t find where output data must be setted

**Comment by dinoslavr** [comment]
Or maybe you can see what MIDI information is sent in this section of the tracks window
![Screenshot 2022-02-06 at 15 14 00](https://user-images.githubusercontent.com/82338269/152682754-64a5e52f-0a39-4240-a0bf-a38f7a4f5c68.png)

**Comment by dinoslavr** [comment]
I don't think you need to set the output data, you just need to find what Logic sends and read it. Try record enabling one track an observe what the window in the screenshot above shows. The same goes for mute, solo, etc.

**Reply by benwadub** [reply]
Logic doesn’t send anything if no control surface is setted. And I didn’t want to use the mcu protocol because of the need to set two cables for my 16 tracks and also the inability to access all the parameters at once like audio track and return track. for ableton live I already made a big script so I d like to not change my code because that was a lot of work for my big controller

**Reply by dinoslavr** [reply]
You're right! I'm afraid my knowledge on the subject is not so extended yet :( I always had a control surface connected so I never faced this issue

**Reply by benwadub** [reply]
All I found as solution is to make a lua script but I can t find any information very clear for the moment, I ll look deeper in this direction.

**Comment by dinoslavr** [comment]
Great! I'm sorry I couldn't be more helpful!

**Reply by benwadub** [reply]
No problem thanks to answered me!

**Comment by dinoslavr** [comment]
You're welcome :)
